### PR TITLE
qubes-core-vchan-xen: init at 4.1.2

### PR DESCRIPTION
**Pull request by 0x4A6F**
###### Motivation for this change
Adding core QubesOS services for better integration.

During CCCamp19 we (@erictapen, +others) sat down with @marmarek and talked about NixOS Integration in QubesOS.

This package is a dependency for qubes-core-qrexec.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by marmarek** [comment]
I haven't uploaded my WIP version yet (I've got half into qrexec packaging), but I did improved makefile in qubes-core-vchan-xen upstream. On camp we've hit various issues about library paths (pkgconfig file, include etc). Now it's possible to put them in /lib /include etc, instead of /usr, with a `PREFIX=/`.

File: pkgs/applications/qubes/qubes-core-vchan-xen/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchFromGitHub, xen_4_10 }:
+
+stdenv.mkDerivation rec {
+  pname = "qubes-core-vchan-xen";
+  version = "4.1.0";
```
Comment (marmarek) [inline pkgs/applications/qubes/qubes-core-vchan-xen/default.nix]: 4.1.2 has said improvements.

File: pkgs/applications/qubes/qubes-core-vchan-xen/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    rev = "v${version}";
+    sha256 = "1p7gc7lyi53zcy9dyikjrkkmfa9873sabypqb7cp4cr4371cl8q6";
+  };
+
+  buildInputs = [ xen_4_10 ];
+  buildPhase = ''
+    make all
+  '';
+  installPhase = ''
+    make install DESTDIR=$out
```
Comment (marmarek) [inline pkgs/applications/qubes/qubes-core-vchan-xen/default.nix]:
```suggestion
    make install DESTDIR=$out PREFIX=/
```
There is also LIBDIR and INCLUDEDIR if needed.

File: pkgs/applications/qubes/qubes-core-vchan-xen/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  src = fetchFromGitHub {
+    owner = "QubesOS";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1p7gc7lyi53zcy9dyikjrkkmfa9873sabypqb7cp4cr4371cl8q6";
+  };
+
+  buildInputs = [ xen_4_10 ];
+  buildPhase = ''
+    make all
```
Comment (marmarek) [inline pkgs/applications/qubes/qubes-core-vchan-xen/default.nix]:
```suggestion
    make all PREFIX=/
```
Prefix needs to be set here too, as it's built into pkgconfig file.

**Review by marmarek — CHANGES_REQUESTED** [review]

**Comment by marmarek** [comment]
And thanks for pushing it here!

File: pkgs/applications/qubes/qubes-core-vchan-xen/default.nix (hunk 1 above)
Comment (0x4A6F) [reply pkgs/applications/qubes/qubes-core-vchan-xen/default.nix]: Thanks, updated with `PREFIX=/`.

File: pkgs/applications/qubes/qubes-core-vchan-xen/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  '';
+  installPhase = ''
+    make install DESTDIR=$out PREFIX=/
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Libraries required for the higher-level Qubes daemons and tools";
+    homepage = "https://qubes-os.org";
+    license = licenses.gpl2Plus;
+    maintainers = with maintainers {}; [ "0x4A6F" ];
```
Comment (Lassulus) [inline pkgs/applications/qubes/qubes-core-vchan-xen/default.nix]:
```suggestion
    maintainers = with maintainers; [ "0x4A6F" ];
```

Comment (0x4A6F) [reply pkgs/applications/qubes/qubes-core-vchan-xen/default.nix]: Thanks.

**Comment by Lassulus** [comment]
So this just seem to be some header files and nothing else? so not much to review then I guess

**Comment by marmarek** [comment]
There should be a library (.so) too.

**Comment by 0x4A6F** [comment]
```
/nix/store/nwjvmbqp3ldzy9hhbcd363sldimhg3s3-qubes-core-vchan-xen-4.1.2
├── include
│   ├── u2mfn-kernel.h
│   ├── u2mfnlib.h
│   └── vchan-xen
│       └── libvchan.h
└── lib
    ├── libu2mfn.so
    ├── libvchan-xen.so
    └── pkgconfig
        └── vchan-xen.pc

4 directories, 6 files
```

**Review by Lassulus — APPROVED** [review]
tested with nix-review

**Pull request merged**